eole/predict/generator.py

```diff
@@ -180,11 +178,14 @@ def _predict_batch_with_strategy(self, batch, decode_strategy, left_pad=True):
             dec_in = torch.cat((src, dec_in), 1)
             tgt_pad_mask = dec_in.eq(self._tgt_pad_idx).unsqueeze(1)  # [B, T_tgt]
             emb = self.model.tgt_emb(dec_in)
+            self.model.decoder._clear_cache()
+            position_embeddings = self.model.rope.update(dec_in.size(1), step=0)
```

Comment:
Same as in inference.py.

Also, not strictly linked to this PR but these duplicated changes lead to the question of the duplicated code -> shouldn't this estimator part be moved up in inference._decode_and_generate?

Reply:
no, it happens after but would may need a super to something that has a different name:
def _predict_batch_with_strategy(self, batch, decode_strategy, left_pad=True):
def _translate_batch_with_strategy(self, batch, decode_strategy):
so for another PR